fastapi_filter/contrib/beanie/filter.py

```diff
@@ -0,0 +1,125 @@
+            count__lte: Optional[int]
+            created_at__gt: Optional[datetime]
+            name__ne: Optional[str]
+            name__nin: Optional[list[str]]
+            name__isnull: Optional[bool]
+        ```
+    """
+
+    @property
+    def filtering_fields(self) -> Iterable[tuple[str, Any]]:
+        """Get filtering fields.
+
+        self.model_dump() serializes ObjectId to str which is not what we want.
+        """
+        return {field_name: getattr(self, field_name) for field_name, _ in super().filtering_fields}.items()
+
+    def sort(self, query: FindMany[FindType]) -> FindMany[FindType]:
+        if not self.ordering_values:
+            return query
+        return query.sort(*self.ordering_values)
+
+    @field_validator("*", mode="before")
+    @classmethod
+    def split_str(
+        cls: type["BaseFilterModel"], value: Optional[str], field: ValidationInfo
+    ) -> Optional[Union[list[str], str]]:
+        if (
+            field.field_name is not None
+            and (
+                field.field_name == cls.Constants.ordering_field_name
+                or field.field_name.endswith("__in")
+                or field.field_name.endswith("__nin")
+            )
+            and isinstance(value, str)
+        ):
+            if not value:
+                # Empty string should return [] not ['']
+                return []
+            return list(value.split(","))
+        return value
+
+    def _get_filter_conditions(self, nesting_depth: int = 1) -> list[tuple[Mapping[str, Any], Mapping[str, Any]]]:
+        filter_conditions: list[tuple[Mapping[str, Any], Mapping[str, Any]]] = []
+        for field_name, value in self.filtering_fields:
+            field_value = getattr(self, field_name)
+            if isinstance(field_value, Filter):
+                if not field_value.model_dump(exclude_none=True, exclude_unset=True):
+                    continue
+
+                filter_conditions.append(
+                    (
+                        {field_name: _odm_operator_transformer["neq"](None)},
+                        {"fetch_links": True, "nesting_depth": nesting_depth},
+                    )
+                )
+                for part, part_options in field_value._get_filter_conditions(nesting_depth=nesting_depth + 1):  # noqa: SLF001
+                    for sub_field_name, sub_value in part.items():
+                        filter_conditions.append(
+                            (
+                                {f"{field_name}.{sub_field_name}": sub_value},
+                                {"fetch_links": True, "nesting_depth": nesting_depth, **part_options},
+                            )
+                        )
+
+            elif "__" in field_name:
+                stripped_field_name, operator = field_name.split("__")
+                search_criteria = _odm_operator_transformer[operator](value)
+                filter_conditions.append(({stripped_field_name: search_criteria}, {}))
```

Comment:
I would like to have `getattr(self.Constants.model, stripped_field_name)` here to make it work with custom attributes here as well

Reply:
@ADR-007 did it

Reply:
Thank you!